test-framework/common/src/main/java/io/quarkus/test/common/configuration/InMemoryConfigSourcePopulator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.test.common.configuration;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+import io.quarkus.runtime.configuration.ExpandingConfigSource;
+
+/**
+ * Class that populates configuration parameters from test to In Memory Config Source
+ */
+public final class InMemoryConfigSourcePopulator {
+
+    public void populate(final Class<?> testClass) {
+
+        final InMemoryConfigSource inMemoryConfigSource = findInMemoryConfigSource();
+        if (inMemoryConfigSource != null) {
+            final ConfigurationPropertiesExtractor configurationPropertiesExtractor = new ConfigurationPropertiesExtractor();
+            inMemoryConfigSource.addProperties(configurationPropertiesExtractor.extract(testClass));
```

Comment:
Are these config values ever cleared after the test? I can't seem to see where that happens

Reply:
True if the JVM is reused, going to fix that.